lib/run/index.js

```diff
@@ -142,6 +142,17 @@ module.exports = function (options, callback) {
             _.isArray(entrypoint.execute) && (entrypoint.lookupStrategy = MULTIENTRY_LOOKUP_STRATEGY);
         }
 
+        if (options.iterationData && (options.iterationDataFrom || options.iterationDataTo)) {
+            let dataSize = options.iterationData.length + 1,
+                start = options.iterationDataFrom,
+                end = options.iterationDataTo;
+
+            start = start >= 1 && start < dataSize ? start : 1;
+            end = end >= start && end <= dataSize ? end : dataSize;
+
+            options.iterationData = options.iterationData.slice(start - 1, end);
```

Comment:
This needs to be done at `runtime` layer and not at Newman CLI layer for portability across usage in app and CLI. 

Additionally, care has to be taken that for ridiculously large data file size (not so uncommon) there isn't a memory usage spike.